ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/UserManagementLimitsIT.scala

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.testtool.suites
+
+import java.util.UUID
+
+import com.daml.error.definitions.LedgerApiErrors
+import com.daml.ledger.api.testtool.infrastructure.Allocation._
+import com.daml.ledger.api.testtool.infrastructure.Assertions._
+import com.daml.ledger.api.testtool.infrastructure.LedgerTestSuite
+import com.daml.ledger.api.v1.admin.user_management_service.{
+  CreateUserRequest,
+  DeleteUserRequest,
+  GrantUserRightsRequest,
+  ListUserRightsRequest,
+  User,
+  Right => Permission,
+}
+import io.grpc.Status
+
+import scala.collection.immutable.Iterable
+
+final class UserManagementLimitsIT extends LedgerTestSuite {
+
+  def assertSameElements[T](actual: Iterable[T], expected: Iterable[T]): Unit = {
+    assert(
+      actual.toSet == expected.toSet,
+      s"Actual ${actual.mkString(", ")} should have the same elements as (expected): ${expected.mkString(", ")}",
+    )
+  }
+
+  def assertEquals(actual: Any, expected: Any): Unit = {
+    assert(actual == expected, s"Actual |${actual}| should be equal (expected): |${expected}|")
+  }
+
+  test(
+    "UserManagementUserRightsLimit",
+    "Test user rights per user limit",
+    allocate(NoParties),
+    enabled = _.userManagement,
+    disabledReason = "requires user management feature",
+  )(implicit ec => { case Participants(Participant(ledger)) =>
+    def assertTooManyUserRightsError(t: Throwable): Unit = {
+      assertGrpcError(
+        participant = ledger,
+        t = t,
+        expectedCode = Status.Code.FAILED_PRECONDITION,
+        selfServiceErrorCode = LedgerApiErrors.AdminServices.TooManyUserRights,
+        exceptionMessageSubstring = None,
+      )
+    }
+
+    def createCanActAs(id: Int) =
+      Permission(Permission.Kind.CanActAs(Permission.CanActAs(s"acting-party-$id")))
+
+    val user1 = User(UUID.randomUUID.toString, "")
```

Comment:
There is a prior art of how we generate such strings that guarantee isolation and at the same time allow us to debug logfiles/database contents when something goes wrong. Please have a look at this and adopt the code:
https://github.com/digital-asset/daml/blob/d4ebce6f4446c9f4744ed610cfb83161a8640d16/ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/participant/ParticipantTestContext.scala#L134

Reply:
Ok